app/scripts/lib/AddressBookPetnamesBridge.ts

```diff
@@ -35,14 +52,10 @@ export class AddressBookPetnamesBridge extends AbstractPetnamesBridge {
     const entries: PetnameEntry[] = [];
     const { state } = this.#addressBookController;
     for (const chainId of Object.keys(state.addressBook)) {
-      // TODO: Replace `any` with type
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const chainEntries = state.addressBook[chainId as any];
+      const chainEntries = state.addressBook[chainId as Hex];
```

Comment:
```suggestion
      if(!isStrictHexString(chainId)){
        continue
      }
      const chainEntries = state.addressBook[chainId];
```
We can avoid the type assertions if we add a type guarding check for chainId as Hex.

Reply:
Good idea, captured here [c1fe3e](https://github.com/MetaMask/metamask-extension/pull/27107/commits/c1fe3ed5b5491de0f17019a58a50afb6512a07f8)

Reply:
```suggestion
      const chainEntries = state.addressBook[chainId];
```
we don't need remove this type assertion after adding the type guarding